pkg/monitor/controller.go

```diff
@@ -146,12 +178,44 @@ func (monitor *controllerMonitor) deleteStaleAgentCRDs() {
 		klog.Errorf("Failed to list agent monitoring CRDs: %v", err)
 		return
 	}
-	// Delete stale agent monitoring CRD based on existing nodes.
-	nodeLister := monitor.nodeInformer.Lister()
+	// Delete stale agent monitoring CRD based on existing Nodes and ExternalNodes.
+	externalNodeNames := sets.NewString()
+	if features.DefaultFeatureGate.Enabled(features.ExternalNode) {
+		externalNodes, err := monitor.externalNodeLister.List(labels.Everything())
```

Comment:
Do you think it is easier if we create two sets: one is for the existsing crds, and the other is for existing Nodes + ExternalNodes ( if feature gate is enabled). Make a diff to get the stale crd name, and then do `monitor.deleteAgentCRD`.

Reply:
Got it, have updated accordingly.